docs/dev/development-process/development-process.rst

```diff
@@ -0,0 +1,209 @@
+In addition, the development of a comprehensive data model is important for a clear understanding of how data is securely stored, retrieved and managed within the system, taking into account both logical and physical data perspectives. 
+Security aspects, especially access control mechanisms, are considered to ensure data integrity and privacy.
+
+Step 5: UI/UX Design
+--------------------
+If the proposed feature requires a change to the software's user interface, this section should include a detailed description of the proposed changes, as well as a mockup that illustrates the new user interface.
+The mockup should be created using Figma in conjunction with the Artemis Design System and should be added to the feature proposal as a screenshot. 
+Furthermore it is important to include a description of the user flow that references the dynamic model created in the analysis section.
+
+.. figure:: ./uiux_workflow.png
+    :align: center
+    :alt: UI/UX Design Workflow - Activity Diagram
+    :figclass: align-center
+
+.. raw:: html
+
+    <div style="text-align: center;">
+        <iframe src="https://live.rbg.tum.de/w/artemisintro/46233?video_only=1&t=0" allowfullscreen="1" frameborder="0" width="600" height="300">
+            Watch this video on TUM-Live.
+        </iframe>
+    </div>
+
+
+4. Implement the Feature
+========================
+In this step, the development team converts the detailed plans and designs outlined in the functional proposal into working code. 
+This step requires careful adherence to the previously defined requirements and system architecture to ensure that the function fits seamlessly into the existing system and fulfills the specified functional and performance criteria.
+
+.. note::
+      Make sure to follow the `Artemis Code and Design Guidelines <https://docs.artemis.cit.tum.de/dev/guidelines.html>`_.
+
+5. Create a Pull Request
+========================
+After the feature implementation is complete, the developer is required to create a pull request for integrating the feature into the develop branch. 
+The subsequent sections provide guidance on the naming conventions and outline the necessary steps for creating and merging a pull request.
+
+Naming Conventions for GitHub Pull Requests
+-------------------------------------------
+
+1. The first term is a main feature of Artemis and should use code highlighting, e.g.  ``Programming exercises``:
+
+    1. Possible feature tags are: ``Programming Exercises``, ``Integrated Code Lifecycle``, ``Quiz Exercises``, ``Modeling Exercises``, ``Text Exercises``, ``File Upload Exercises``, ``Exam Mode``,
+       ``Grading``, ``Assessment``, ``Communication``, ``Notifications``, ``Team Exercises``, ``Lectures``, ``Integrated Markdown Editor``, ``Plagiarism Checks``, ``Learning Analytics``,
+       ``Adaptive Learning``, ``Learning Path``, ``Tutorial Groups``, ``Iris``, ``Scalability``, ``Usability``, ``Performance``, ``Intrastructure``, ``Mobile Apps``.
```

Comment:
Please revert the changes to the capitalization, as we do not capitalize terms after the first word (even if this is not according to english title capitalization)
```suggestion
    1. Possible feature tags are: ``Programming exercises``, ``Integrated code lifecycle``, ``Quiz exercises``, ``Modeling exercises``, ``Text exercises``, ``File upload exercises``, ``Exam mode``,
       ``Grading``, ``Assessment``, ``Communication``, ``Notifications``, ``Team exercises``, ``Lectures``, ``Integrated markdown editor``, ``Plagiarism checks``, ``Learning analytics``,
       ``Adaptive learning``, ``Learning path``, ``Tutorial groups``, ``Iris``, ``Scalability``, ``Usability``, ``Performance``, ``Intrastructure``, ``Mobile apps``.
```

Reply:
Can you double-check the capitalization as mentioned by @rstief?